log_cache_test.go

```diff
@@ -86,3 +89,60 @@ func TestLogCache(t *testing.T) {
 		t.Fatalf("err: %v", err)
 	}
 }
+
+type errorStore struct {
+	LogStore
+	mu      sync.Mutex
+	fail    bool
+	failed  int
+	failMax int
+}
+
+func (e *errorStore) StoreLogs(logs []*Log) error {
+	e.mu.Lock()
+	defer e.mu.Unlock()
+	if e.fail {
+		e.failed++
+		if e.failed <= e.failMax {
+			return errors.New("some error")
+		}
+		e.fail = false
+	}
+	return e.LogStore.StoreLogs(logs)
+}
+
+func (e *errorStore) failNext(count int) {
+	e.mu.Lock()
+	e.fail = true
+	e.failMax = count
+	e.mu.Unlock()
+}
+
+func TestLogCacheWithBackendStoreError(t *testing.T) {
+	var err error
+	store := NewInmemStore()
+	errStore := &errorStore{LogStore: store}
+	c, _ := NewLogCache(16, errStore)
+
+	for i := 0; i < 4; i++ {
+		log := &Log{Index: uint64(i) + 1}
+		store.StoreLog(log)
+	}
+	errStore.failNext(1)
+	log := &Log{Index: 5}
+	err = c.StoreLog(log)
+	if !strings.Contains(err.Error(), "unable to store logs in log store") {
+		t.Fatalf("Should have returned unable to store logs in log store,  got err=%v", err)
+	}
```

Comment:
may need to update this after removing the wrapping: it's failing in CI: https://app.circleci.com/pipelines/github/hashicorp/raft/343/workflows/82684bfa-24ff-4bd7-91f2-17afb2efccc1/jobs/5360